File: POO_Python/Persona.py
# Definición de una clase llamada 'Persona'
class Persona:
    # Método constructor, se llama automáticamente al crear un nuevo objeto de la clase
    def __init__(self, nombre, edad, altura):
        self.nombre = nombre
        self.edad = edad
        self.altura = altura

    # Metodo de persona
    def bailar(self):
        return f"{self.nombre} esta bailando"
    
    def comer(self):
        return f"{self.nombre} estoy comiendo"
    
    def caminar(self):
        return f"{self.nombre} voy a caminar"
    
    # Método para obtener información de la persona
    def obtener_informacion(self):
        return f"Nombre: {self.nombre}, Edad: {self.edad}, Altura: {self.altura}"

# Crear objetos (instancias) de la clase Persona
persona = Persona("Jesus Antonio", 25, 1.83)
persona2 = Persona("María", 30, 1.70)

# Acceder a los atributos de los objetos
print(persona.nombre)
print(persona2.nombre, persona2.edad)

# Llamar a un método de la clase
info_persona = persona.obtener_informacion()
print(info_persona)

# Llamer metodos de persona
bailar_persona = persona.bailar()
print(bailar_persona)

comer_perosna = persona.comer()
print(comer_perosna)

caminar_persona = persona.caminar()
print(caminar_persona)

File: POO_JavaScript/Persona.js
// Definición de una clase llamada "Persona"
class Persona {
    // Constructor para inicializar la clase con datos iniciales
    constructor(nombre, edad, altura) {
      this.nombre = nombre;
      this.edad = edad;
      this.altura = altura;
    }
  
    // Método para obtener la información de la persona
    obtenerInformacion() {
      return `Nombre: ${this.nombre}, Edad: ${this.edad} años, Altura :${this.altura} m`;
    }
  
    // Método para saludar
    saludar() {
      console.log(`¡Hola! Mi nombre es ${this.nombre}`);
    }
    // Metodos de Persona
    bailar() {
      console.log(`${this.nombre} esta bailando`);
    }

    comer() {
      console.log(`${this.nombre} estoy comiendo`);
    }

    caminar() {
      console.log(`${this.nombre} voy caminando`)
    }
  }
  
  // Crear una instancia de la clase Persona
  const persona = new Persona('Jesus Antonio', 25, 1.82);
  
  // Acceder a los métodos y propiedades de la instancia
  console.log(persona.obtenerInformacion());
  persona.saludar();
  persona.bailar();
  persona.comer();
  persona.caminar()